internal/controller/nodestate_cleaner_controller.go

```diff
@@ -96,3 +98,19 @@ func (r *NodeStateCleaner) SetupWithManager(mgr ctrl.Manager) error {
 		}).
 		Complete(r)
 }
+
+type PodSelector struct {
+	Key   string
+	Value string
+}
+
+func PodSelectorFromString(s string) (PodSelector, error) {
```

Comment:
The string can finally be not valid k8s labels but that function will pass (and probably will just fail on the list).

https://stackoverflow.com/questions/75081380/generated-valid-label-value-kubernetes (random link)

Reply:
valid point, added